src/app/resources.py

```diff
@@ -77,198 +86,200 @@ def delete(self, *args, **kwargs):
-    'username': fields.String,
-    'password': fields.String,
-    'first_name': fields.String,
-    'last_name': fields.String,
-    'phone': fields.Integer,
-})
+@api.representation('/json')
+class Logout(Resource):
+    """
+    Logout resource.
+    Remove the username from the session.
+    """
 
+    def get(self):
+        sess.pop('email', None)
+        return 'Dropped!', 200  # OK
 
-@api.representation('/json')
+
+# @api.representation('/json')
 class Register(Resource):
     """
     Register resource.
-
     Parsing requested data, checks if username or email doesn't exit in DB, then create user in DB. 200 OK
-    Otherwise, return 500 or 400 error
+    Otherwise, return 500 or 400 error.
     """
 
-    @api.expect(user_model)
+    @api.expect(user_post)
     def post(self):
-        parser = reqparse.RequestParser()
-        parser.add_argument('email', type=str, help='')
-        parser.add_argument('username', type=str, help='')
-        parser.add_argument('password', type=str, help='')
-        parser.add_argument('first_name', type=str, help='')
-        parser.add_argument('last_name', type=str, help='')
-        parser.add_argument('phone', type=str, help='')
-        args = parser.parse_args()
-        # json_data = request.get_json()
-        # if not json_data or not isinstance(json_data, dict):
-        #     return {'message': 'No input data provided'}, 400  # Bad Request
+
+        json_data = request.get_json()
+        if not json_data or not isinstance(json_data, dict):
+            return 'No input data provided', 400  # Bad Request
 
         # Validate and deserialize input
         try:
-            # data = UserSchema().load(json_data)
-            data = UserSchema().load(args)
+            data = UserSchema().load(json_data)
         except ValidationError as err:
-            return str(err), 422  # Unprocessable Entity
+            return err, 422  # Unprocessable Entity
 
+        # TODO One function for all
         # Check if a new user is not exist in data base
         if session.query(User).filter(User.username == data['username']).first():
-            msg = "User with username: '{0}' is ALREADY EXISTS.".format(data['username'])
+            msg = f"User with username: {data['username']} is ALREADY EXISTS."
             return msg, 200  # OK
         elif session.query(User).filter(User.email == data['email']).first():
-            msg = "User with email: '{0}' is ALREADY EXISTS".format(data['email'])
+            msg = f"User with email: {data['email']} is ALREADY EXISTS"
             return msg, 200  # OK
         else:
-            # TODO optimization USER CLASS
-            # user = User(data)
-            user = User(data['username'], data['email'], data['password'],
-                        data['first_name'], data['last_name'], data['phone'])
-
             # crate a new user
             try:
-                session.add(user)
+                session.add(User(data))
```

Comment:
I think you should leave previsous statement, cause User(data) doesn't seem to be right statement. Or you shold right something like: sesson.add(data), cause I think data is already User model? What do you think?

Reply:
Or tell me what the return value of UserSchema().load?